kobuki_keyop/src/keyop_node.cpp

```diff
@@ -27,42 +27,23 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-/**
- * @file /kobuki_keyop/src/main.cpp
- *
- * @brief Executable code for the keyop node.
- *
- **/
+#include <cstdio>
+#include <memory>
 
-/*****************************************************************************
-** Includes
-*****************************************************************************/
+#include <rclcpp/rclcpp.hpp>
 
-#include "../include/keyop_core/keyop_core.hpp"
+#include "kobuki_keyop/keyop.hpp"
 
-/*****************************************************************************
-** Using
-*****************************************************************************/
+int main(int argc, char** argv)
+{
+  // Force flush of the stdout buffer.
+  setvbuf(stdout, nullptr, _IONBF, BUFSIZ);
 
-using keyop_core::KeyOpCore;
+  rclcpp::init(argc, argv);
 
-/*****************************************************************************
-** Main
-*****************************************************************************/
+  rclcpp::spin(std::make_shared<kobuki_keyop::KeyOp>(rclcpp::NodeOptions()));
```

Comment:
Maybe a rclcpp noob question - how does this get to Keyop's `spin()` method?

Reply:
Through the executor.  `KeyOp::spin` is now a wall timer, so every 100 milliseconds, it will place a work item onto the executor queue.  `rclcpp::spin` will then pick up that work item when it is available.

That does bring up the good question of how that work item gets onto the queue.  But that goes deeper into how the executor works than I know, so I don't know if the executor just periodically wakes up to check for work, uses a thread for these things, or does something else.

Reply:
Aha, missed that when scanning for it. Annoying how often github hides diffs.

Should Keyop's spin method lock the mutex? I' It's going to be fighting with the input thread for those variables.

Do we need to know about how that work item gets onto the queue? I could make guesses, but  I don't think that's going to cause us problems here.

Reply:
> Should Keyop's spin method lock the mutex? I' It's going to be fighting with the input thread for those variables.

Yep, it absolutely show lock cmd_mutex while executing.  Good catch, fixed in 75d43e1